keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,730 @@
+
+- There needs to be a way for the Kubelet to verify the CRI provider is capable of providing the correct metrics. Upon upgrading to a version that relies on this new behavior (assuming the feature gate is enabled), Kubelet should fail early if the CRI implementation won't report the expected metrics.
+- For Beta/GA releases, components that rely on `/metrics/cadvisor` should take the decided action (use `/stats/summary`, or use the Kubelet provided `/metrics/cadvisor`).
+
+### Version Skew Strategy
+
+- Consumers of `/metrics/cadvisor` will have at least `n-1` versions to take the decided action on using `/metrics/cadvisor`.
+    - This may include prometheus, kubectl top plugin
+- In general, CRI should be updated in tandem with or before the Kubelet.
+- Kubelet should handle CRI not having the correct version.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: PodAndContainerStatsFromCRI
+    - Components depending on the feature gate: Kubelet, cAdvisor
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+  Enabling this behavior means some stats endpoints will not be filled:
+  - some entries in `/metrics/cadvisor`
+  - Accelerator and UserDefinedMetrics in `/stats/summary`
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, assuming the Kubelet is restarted.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  There should be no problem with this.
+
+* **Are there any tests for feature enablement/disablement?**
+  Should not be necessary,
```

Comment:
What happens if you disable the feature gate on a live kubelet? Will you manually test this?